packages/gatsby/src/internal-plugins/functions/gatsby-node.ts

```diff
@@ -372,6 +399,25 @@ export async function onPreBootstrap({
           reporter.error(formatted.errors)
         }
 
+        // After the production build, we scan compiled functions for NPM
+        // dependencies and add those to the manifest.json to aid deployment
+        // packaging.
+        if (isProductionEnv) {
+          const pathToManifest = path.join(
+            compiledFunctionsDir,
+            `manifest.json`
+          )
+          const manifest = JSON.parse(fs.readFileSync(pathToManifest, `utf-8`))
+
+          manifest.forEach((func, i) => {
```

Comment:
after a build, loop through functions and add any dependencies they have to the manifest.json